trie/branchNode.go

```diff
@@ -421,6 +439,25 @@ func (bn *branchNode) resolveCollapsed(pos byte, db common.DBWriteCacher) error
 	return nil
 }
 
+func (bn *branchNode) resolveCollapsedFromBytes(pos byte, childNode []byte) error {
```

Comment:
this is almost the resolveCollapsedFromDb function. The only difference is that the resolveCollapsedFromDb takes the data from DB rather than provided bytes. Let's extract the common code. Valid for all node types.

Reply:
Refactored.